### ISSUE TITLE
Add optional rounding for more than just `px` values

### ISSUE THREAD

**Issue by maranomynet** [jakubpawlowicz/clean-css#686]
When using Stylus to calculate lots of `em`, `rem`and `%` values one ends up with **tons** of ridiculously precise (15 digit decimal fractions) numbers.

Clean-css should also allow me to cap the precision of these values – independently of the `--rounding--precision` option which only applies to `px` values.

**Comment by jakubpawlowicz** [comment]
Sure, that can be easily done :+1:  Can you think of a sensible default precision for each of those units?

**Comment by maranomynet** [comment]
If you want to be super cautious don't rounding at all by default.

Personally, I'd set 6 as my overall default, as I guess I would be more than sufficient for everything but the most weird/contrived edge-cases...

I think you shouldn't risk going any more fine-grained than that - except perhaps for the physical units like `px`, `mm`, etc. but hardly anyone uses those anyway.

Per-unit-precision options probably won't be commonly used - but they'd be very nice to have.

(still thinking about this...)

**Comment by jakubpawlowicz** [comment]
Hey @maranomynet. It's been a while but, since it's gonna be a breaking change, I'm going to make it in the upcoming 4.0 release. I'm considering:

1. disabling rounding by default
2. allowing non-px units to be rounded by using `{ px: 6, em: 3 }` (API) or `px:6,em:3` (CLI).

**Comment by maranomynet** [comment]
> "except perhaps for the physical units like `px`, `mm` ..."

BTW, that was a typo - I meant `pt`, not `px`. (Points is a non-SI length unit, equal to 0.352778mm)

When it comes to CSS units, `px` is the odd one out really - because all the other units end up being  translated to px upon rendering, but `px` values always remain as `px`. So there's less need for super-precise `px` values.

I'd like to have the option of setting a rounding-precision to all units at once (excluding maybe `px` for the reason mentioned above).

Having to list the desired precision for each and every unit every time would be annoying - since in vast majority of cases you'd want the same ballpark precision for all of them.

**Comment by jakubpawlowicz** [comment]
Sure, I actually read `pt` there ;-)

Re setting desired precision, how about sth like `{ '*': 3, px: 6 }` or `*:3,px:6` means 3 is default for all units while 6 is for `px`?

**Comment by maranomynet** [comment]
Sounds very nice.

**Comment by jakubpawlowicz** [comment]
Cool!

**Comment by vlakoff** [comment]
Your asterisk suggestion sounds very nice. Some examples:

    --rounding *:6       // round everything to 6
    --rounding 6         // shorthand, equivalent to above
    --rounding px:3      // round only px to 3
    --rounding *:6,px:3  // round everything to 6, except px to 3
    --rounding *:6,em:-1 // round everything to 6, disable for em

I just hope it's not being too complicated to implement.

**Comment by jakubpawlowicz** [comment]
👍 It should work exactly like that. Since asterisk has to be quoted in CLI (as with compatibility options) we may want to use `all` as an alias, e.g `--rounding all:6,px:3`.

It shouldn't be too hard to implement as sth similar is already done for compatibility.

**Comment by jakubpawlowicz** [comment]
Have a look at 21679e76ed8810f480dd481144bb3dbe7dffbef9 if you have a moment. 

Have I covered all units?
Should `%` be on the list too?

**Comment by maranomynet** [comment]
Yes, `%` should absolutely be on the list.

**Comment by maranomynet** [comment]
And the list seems exhaustive.

**Comment by jakubpawlowicz** [comment]
Thanks @maranomynet for checking. I'll push it to master and if anything is missing before we can still adjust it before 4.0 release.
